plugins/UsersManager/API.php

```diff
@@ -264,6 +267,64 @@ private function getDefaultUserPreference($preferenceName, $login)
         }
     }
 
+    /**
+     * Returns all users with their role for $idSite.
+     *
+     * @param int $idSite
+     * @param int|null $limit
+     * @param int|null $offset
+     * @param string|null $filter_search text to search for in the user's login, email and alias (if any)
+     * @param string|null $filter_access only select users with this access to $idSite. can be 'noaccess', 'some', 'view', 'admin', 'superuser'
+     *                                   Filtering by 'superuser' is only allowed for other superusers.
+     * @return array
+     */
+    public function getUsersPlusRole($idSite, $limit = null, $offset = 0, $filter_search = null, $filter_access = null)
+    {
+        if (!$this->isUserHasAdminAccessTo($idSite)) {
+            // if the user is not an admin to $idSite, they can only see their own user
+            if ($offset > 1) {
+                return [
+                    'total' => 1,
```

Comment:
This is something we probably should have in most API requests but currently don't have in any (`total`, `results`, `currentPage`, ...). Using this here now kind of means we will need to be using same wording etc in all API requests and probably prevents filters from limit/offset etc to be working correctly? Is this needed for paging? Any chance we could fetch all users in the UI for now and do the paging in JS (assuming it is still fast for 20k users)?  

Or maybe we could apply this more central so it works for more APIs? Like usually we would also fetch all results... then the limit/offset is applied in generic filter and we would know the number of total results etc.

It would be good to have a flag to enable this as a global feature similar to processed reports where we add various meta data.

Reply:
In this case the limit/offset is in the SQL, which may be hard to translate to other API methods automatically.

As for loading everything in JS, if a user searches and finds 10000 users & updates the access for all of them, that would mean sending 10000 user logins to an API method. I'm not sure if that would pose a problem, but it doesn't seem like it would scale.

If the issue is just the wording in the response of these new API methods (I didn't change the output of other methods), maybe we can make them unsupported API?

Btw, `currentPage` wouldn't apply here since the paging is controlled by the JS, this is just for limiting/offsetting the results & still getting the total available result count.

Reply:
As discussed on slack will try using HTTP headers or just hiding the new API methods.